### DISCUSSION TITLE
Customized pocketbase builder

### DISCUSSION THREAD

**Discussion by kennethklee** [pocketbase/pocketbase#2084] [Show and tell]
tldr; build pocketbase with plugins: github.com/kennethklee/xpb

It occurred to me that pocketbase is architected in a way that is easily extensible.

I'm using PB for a personal project and wanted a way to incorporate plugins for different use cases and deployments. After figuring out a way, thought I'd strip plugin functionally down to the bare minimum so others could maybe use.

So here's my quick throw together and attempt at a simple, customized pocketbase builder with plugin support.

It's not very expansive, but it works. Should work with every version of pocketbase going forward. Even created plugins from examples/base code -- timeout, migrations, static dir -- all be able to rebuild examples/base.

feel free to take it or rip it apart piecemeal and make it your own.
https://github.com/kennethklee/xpb

**Comment by jimafisk** [comment]
Hi @kennethklee - I just wanted to drop a note that you helped inspire what we're working on over at https://github.com/pocketbuilds/xpb. If you're still working in this space and want to collab at all, feel free to hit me up. Thanks for sharing your project!

**Comment by kennethklee** [comment]
thanks for the respectful note @jimafisk. unfortunately, I'm not in the space anymore, though can let you know if that changes.
